androidplot-core/src/main/java/com/androidplot/util/FastNumber.java

```diff
@@ -64,6 +64,32 @@ public double doubleValue() {
         return doublePrimitive;
     }
 
+    /**
+     * To be equal, two instances must both be instances of {@link FastNumber}. The inner {@link
+     * #number} field must also be a common type. Numbers which are mathematically equal are not
+     * necessarily equal. This keeps with the java implementation of common Number classes where for
+     * instance {@code new Integer(0).equals(new Double(0))} returns {@code false}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FastNumber that = (FastNumber) o;
+
+        return number != null ? number.equals(that.number) : that.number == null;
+
+    }
+
+    @Override
+    public int hashCode() {
+        return number != null ? number.hashCode() : 0;
```

Comment:
I've disallowed null `FastNumbers` I didn't dig too deeply though as adding nullity annotations to the entire project can be a real rabbit hole and is probably out of scope of this pull request.